### PR TITLE
feat: new facet widgets for "authors", "content types" and "languages"

### PR DESCRIPTION
**Pull request by pirhoo**
## PR description

This PR adds 3 news widgets to explore value facets for "authors", "content type" and "language". All those widgets use the same `WidgetFieldFacets` class and display an horizontal
bar chart next to the value (see https://github.com/ICIJ/datashare/issues/1226).

## Changes

* feat: new  `WidgetFieldFacets` class and component
* feat: remove `card-header` from all widgets for lighter display
* feat: narrow the spacing between widgets

## Preview

Screenshot from 2023-10-19 11-40-25

### CODE REVIEW

File: src/components/widget/WidgetNames.vue [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default {
   },
   computed: {
     total() {
-      return Math.pow(123, 3) + sum(values(this.entities))
```
Comment (caro3801) [inline src/components/widget/WidgetNames.vue:47]: one question: why was it here?

File: src/components/widget/WidgetFieldFacets.vue [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    itemComponent({ href = null } = {}) {
+      return href ? 'a' : 'div'
+    }
+  },
+  props: {
+    /**
+     * The widget definition object.
+     */
+    widget: {
+      type: Object
```
Comment (caro3801) [inline src/components/widget/WidgetFieldFacets.vue:65]:
```suggestion
      type: Object,
      default:()=>{}
```

File: src/components/widget/WidgetFieldFacets.vue [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+  async mounted() {
+    await this.loadFirstPage()
+  },
+  methods: {
+    clearPages() {
+      this.pages.splice(0, this.pages.length)
+    },
+    async loadFirstPage() {
+      this.clearPages()
+      this.loadPageWithLoader()
```
Comment (caro3801) [inline src/components/widget/WidgetFieldFacets.vue:134]:
```suggestion
     await this.loadPageWithLoader()
```

File: src/components/widget/WidgetFieldFacets.vue [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    },
+    reachedTheEnd() {
+      return get(this, 'lastPage.length', 0) < this.bucketsSize
+    },
+    useInfiniteScroll() {
+      return this.offset > 0 && !this.reachedTheEnd
+    }
+  },
+  async mounted() {
+    await this.loadFirstPage()
```
Comment (caro3801) [inline src/components/widget/WidgetFieldFacets.vue:126]:
```suggestion
  mounted() {
    return this.loadFirstPage()
```

**Review by caro3801 — APPROVED** [review]
I like the "facet" concept. Looks great, comments were helpful. Thanks

File: src/components/widget/WidgetNames.vue (hunk 1 above)
Comment (pirhoo) [reply src/components/widget/WidgetNames.vue:47]: Oups, it was a placeholder!

File: src/components/widget/WidgetFieldFacets.vue (hunk 3 above)
Comment (pirhoo) [reply src/components/widget/WidgetFieldFacets.vue:134]: Nice catch.

**Pull request merged**